editor/src/clj/editor/game_object_non_editable.clj

```diff
@@ -47,38 +53,113 @@
+        ext->resource-type (workspace/get-resource-type-map workspace :non-editable)]
+    (into (delete-connected-nodes-tx-data basis self old-sources-input-label)
+          (mapcat (fn [[data]]
+                    (add-resource-node-fn self data ext->resource-type project)))
+          (sort-by val new-value))))
+
+(defn proj-path->index-setter [evaluation-context self new-value old-sources-input-label resource-connections]
+  (let [basis (:basis evaluation-context)
+        project (project/get-project basis self)]
+    (into (disconnect-connected-nodes-tx-data basis self old-sources-input-label resource-connections)
+          (mapcat (fn [[proj-path]]
+                    (:tx-data (project/connect-resource-node evaluation-context project proj-path self resource-connections))))
+          (sort-by val new-value))))
+
+(g/defnode ComponentHostResourceNode
+  (inherits resource-node/NonEditableResourceNode)
 
   (property embedded-component-resource-data->index g/Any
             (dynamic visible (g/constantly false))
             (set (fn [evaluation-context self _old-value new-value]
-                   (let [basis (:basis evaluation-context)
-                         project (project/get-project basis self)
-                         workspace (project/workspace project)
-                         ext->embedded-component-resource-type (workspace/get-resource-type-map workspace :non-editable)]
-                     (-> []
-                         (into (comp (map gt/source-id)
-                                     (distinct)
-                                     (mapcat g/delete-node))
-                               (ig/explicit-inputs basis self :embedded-component-build-targets))
-                         (into (mapcat #(add-embedded-component-resource-node self (key %) ext->embedded-component-resource-type project))
-                               (sort-by val new-value)))))))
+                   (data->index-setter evaluation-context self new-value :embedded-component-build-targets add-embedded-component-resource-node))))
+
+  (property referenced-component-proj-path->index g/Any
+            (dynamic visible (g/constantly false))
+            (set (fn [evaluation-context self _old-value new-value]
+                   (proj-path->index-setter evaluation-context self new-value :referenced-component-build-targets referenced-component-connections))))
 
   (input embedded-component-build-targets g/Any :array :cascade-delete)
-  (output embedded-component-build-targets g/Any :cached (g/fnk [embedded-component-build-targets]
-                                                           (into []
-                                                                 (map (comp #(assoc % :resource (bt/make-content-hash-build-resource %))
-                                                                            util/only-or-throw))
-                                                                 embedded-component-build-targets))))
```

Comment:
This output implementation was moved to `produce-embedded-component-build-targets` above, and the output declaration itself is now one among many outputs just below here.